### PR TITLE
fix(post-merge-cleanup): detect team-prefixed issue identifiers (e.g. lh-97, rad-42)

### LINKED ISSUE #341: fix(post-merge-cleanup): script does not parse team-prefixed issue identifiers (e.g. lh-97, rad-42)
## Observed

After merging 5 `implementation-plan/*` branches (e.g. `implementation-plan/lh-97-buscar-propiedades-vista-tabla`), `post-merge-cleanup.sh` logged:

> No issue number detected in branch name 'implementation-plan/lh-97-buscar-propiedades-vista-tabla', skipping issue close and tracker update.

The script detects numeric issue identifiers (e.g. `#97` or bare `97`) but not team-prefixed Linear-style identifiers like `lh-97`, `rad-42`, or `lh-92`. This is the standard branch naming convention for Linear-integrated repos, where branches are named `feature/lh-98-slug` or `implementation-plan/lh-97-slug`.

## Impact

For any Linear-tracked downstream, the cleanup script never auto-updates issue status after an `implementation-plan/*` or `feature/lh-*` merge. All post-merge tracker transitions must be applied manually by the Portfolio Orchestrator, which is error-prone and contradicts the "automatic tracker update" contract described in the script's own output.

## Proposed fix

Extend the issue-number detection regex in `post-merge-cleanup.sh` to also match `<team>-<number>` patterns (case-insensitive, where `<team>` is 2–6 alpha characters). Extract the numeric part for numeric-only trackers, or pass the full identifier (e.g. `lh-97`) to Linear-aware tracker update calls.

Example patterns to match:
- `feature/lh-98-reducir-initial-bundle` → issue `lh-98`
- `implementation-plan/lh-97-buscar-propiedades-vista-tabla` → issue `lh-97`
- `fix/rad-42-something` → issue `rad-42`

## Source

Retrospective analysis of batch merge of PRs #189–#194 (lhpaul/leasity-exchanges-prototype), batch run 2026-04-27.

### PR DESCRIPTION
**Pull request by lhpaul**
## Summary

Fixes #341.

`post-merge-cleanup.sh` only matched pure numeric issue IDs (e.g. `fix/42-slug`), causing it to log "No issue number detected" for Linear-style team-prefixed branches like `implementation-plan/lh-97-buscar-propiedades-vista-tabla`. After this fix, the script correctly extracts both numeric and team-prefixed identifiers and runs the tracker status update and issue close flow for all matched branches.

## Changes

- **`scripts/development-workflow/post-merge-cleanup.sh`**: Extended regex matching to detect `<team>-<number>` identifiers (2–6 alpha chars, case-insensitive). Added `ISSUE_IDENTIFIER`, `ISSUE_ID_TYPE` variables. The numeric part (`ISSUE_NUMBER`) is used for `gh issue view/close` and `update_tracker_status_best_effort` (which uses `--argjson` / integer matching against GitHub Projects). The full identifier is captured in `ISSUE_IDENTIFIER` for logging and future tracker helpers.
- **`CHANGELOG.md`**: Added entry under `[Unreleased] Fixed`.

## Test coverage

Regex logic verified with 15 unit-style test cases covering:
- Original numeric patterns (still pass)
- Team-prefixed patterns: `lh-97`, `rad-42`, `PROJ-101`, `ab-5` across all branch types
- Edge cases that must NOT match: 1-char prefix, 7-char prefix, no trailing number, unknown prefix

### CODE REVIEW

**Comment by lhpaul** [comment]
### Step 7a Internal Review Gate Summary

**Effective reviewer set**: claude
**Skipped reviewers**: codex (unreachable from Claude Code subagent)
**Verdict**: APPROVED

All reachable internal reviewers approved. The fix correctly extends the issue-number detection regex to match team-prefixed identifiers (2–6 alpha chars + dash + digits). All 15 regex test cases pass. `ISSUE_NUMBER` (numeric part only) is correctly used for `gh issue` and `update_tracker_status_best_effort` calls. CHANGELOG entry is clean (markdownlint passes, no trailing whitespace, no duplicate headers).

Note: codex was unreachable from the current runner — reviewer coverage was reduced from 2 to 1. Human reviewers may re-run Step 7a from a Codex-capable runner if full coverage is required.

**Comment by lhpaul** [comment]
### Automated Fix: commit `15bd1b5`

Addressed **2** finding(s) from cycle 1:

| # | Platform | File | Description |
|---|----------|------|-------------|
| 1 | devin | `scripts/development-workflow/post-merge-cleanup.sh:206,211` | Restored `#$ISSUE_NUMBER` prefix in spec/plan echo messages (numeric IDs had lost `#`) |
| 2 | devin | `CHANGELOG.md:38` | Updated CHANGELOG to accurately state that numeric part is used for `gh issue` and tracker helper calls; full identifier captured for logging only |

<details><summary>Remaining open findings: 0</summary>

All findings resolved.

</details>

File: scripts/development-workflow/post-merge-cleanup.sh [hunk 1]
```diff
@@ -158,12 +203,12 @@ if [ -n "$ISSUE_NUMBER" ]; then
   elif [ "$BRANCH_TYPE" = "spec" ]; then
     # spec/* branches reference the issue but must not close it — the issue stays
     # open for the next workflow stage (writing the implementation plan).
-    echo "Spec branch for issue #$ISSUE_NUMBER merged; issue stays open for the next workflow stage. Updating tracker status to Spec Ready..."
+    echo "Spec branch for issue $ISSUE_IDENTIFIER merged; issue stays open for the next workflow stage. Updating tracker status to Spec Ready..."
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply scripts/development-workflow/post-merge-cleanup.sh]: Fixed in commit `15bd1b5`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -35,6 +35,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- **Team-prefixed issue identifiers not detected in `post-merge-cleanup.sh`** (#341): The script's regex only matched pure numeric IDs (e.g. `fix/42-slug`), causing it to log "No issue number detected" and skip tracker updates for branches using Linear-style team-prefixed identifiers (e.g. `implementation-plan/lh-97-buscar-propiedades-vista-tabla`, `fix/rad-42-something`). Extended the regex to also match `<team>-<number>` patterns (2–6 alpha chars, case-insensitive). `gh issue` commands continue to use the numeric part (`#97`) for GitHub-backed repos; the full identifier (e.g. `lh-97`) is passed to the tracker update helper for Linear-aware downstream projects.
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply CHANGELOG.md]: Fixed in commit `15bd1b5`.

**Comment by lhpaul** [comment]
### Automated Reviewer Loop Summary

**Result:** clean
**Cycles:** 1 / 10
**Platforms:** devin

| # | Platform | File | Line | Description | Status | Resolved In |
|---|----------|------|------|-------------|--------|-------------|
| 1 | devin | `scripts/development-workflow/post-merge-cleanup.sh` | 206 | Echo messages drop `#` prefix for numeric IDs in spec/plan branches | Resolved | `15bd1b5` |
| 2 | devin | `CHANGELOG.md` | 38 | CHANGELOG incorrectly stated full identifier passed to tracker helper | Resolved | `15bd1b5` |

**Resolved:** 2 / 2 findings
